### ISSUE TITLE
Type definition for cells() function confusing

### ISSUE THREAD

**Issue by aaronbeall** [handsontable/handsontable#5599]
### Description

The type definition for the `cells()` function currently looks like this:

```
cells?: (row?: number, col?: number, prop?: object) => GridSettings;
```

### Issues

1. Why are all the arguments marked as optional? This would indicate that sometimes they are `undefined`. This doesn't seem to be true, aren't they always populated?
2. Why is `prop` typed as `object`? Shouldn't that be `string` or `string | number`?
3. Is returning a `GridSettings` object always necessary? Or can `void` be returned?

### Demo

n/a

### Your environment
* Handsontable version:
* Browser Name and version:
* Operating System:

**Comment by AMBudnik** [comment]
Thank you for pointing that out @aaronbeall 

According to our docs

> Possible values of prop:
> * property name for column's data source object, when dataset is an array of objects
> * the same number as col, when dataset is an array of arrays
> Default Value: undefined

so it should be a `string` or a `number`.


To be confirmed by @swistach or @wojciechczerniak 
1. According to docs
> Parameters row and col always represent physical indexes.

we should not have a situation when the value is `undefined`.

3. As far as I know we always return the settings, so `void` is not an option

**Comment by AMBudnik** [comment]
Hey @aaronbeall 

Should that issue be closed with #5767 ?

**Comment by aaronbeall** [comment]
Yep!
